forge/test/vault/ChainVaultsTest.t.sol

```diff
@@ -0,0 +1,237 @@
+
+        console.log("Getting user more want.");
+        deal(vault.want(), address(user), wantStartingAmount);
+        console.log("Approving more want.");
+        user.approve(address(want), address(vault), wantStartingAmount);
+        
+        // Users can't deposit.
+        console.log("Trying to deposit while panicked.");
+        vm.expectRevert("Pausable: paused");
+        user.depositAll(vault);
+        
+        // User can still withdraw
+        console.log("User withdraws all.");
+        user.withdrawAll(vault);
+
+        uint256 wantBalanceFinal = want.balanceOf(address(user));
+        assertGt(wantBalanceFinal, wantStartingAmount * 99 / 100, "Expected wantBalanceFinal > wantStartingAmount * 99 / 100");
+    }
+
+    function test_multipleUsers() external {
+        _unpauseIfPaused();
+        
+        _depositIntoVault(user);
+
+        // Setup second user.
+        VaultUser user2 = new VaultUser();
+        console.log("Getting want for user2.");
+        deal(address(want), address(user2), wantStartingAmount);
+
+        uint256 pricePerFullShare = vault.getPricePerFullShare();
+
+        skip(delay);
+
+        console.log("User2 depositAll.");
+        _depositIntoVault(user2);
+        
+        uint256 pricePerFullShareAfterUser2Deposit = vault.getPricePerFullShare();
+
+        skip(delay);
+
+        console.log("User1 withdrawAll.");
+        user.withdrawAll(vault);
+
+        uint256 user1WantBalanceFinal = want.balanceOf(address(user));
+        uint256 pricePerFullShareAfterUser1Withdraw = vault.getPricePerFullShare();
+
+        assertGe(pricePerFullShareAfterUser2Deposit, pricePerFullShare, "Expected pricePerFullShareAfterUser2Deposit >= pricePerFullShare");
+        assertGe(pricePerFullShareAfterUser1Withdraw, pricePerFullShareAfterUser2Deposit, "Expected pricePerFullShareAfterUser1Withdraw >= pricePerFullShareAfterUser2Deposit");
+        assertGt(user1WantBalanceFinal, wantStartingAmount * 99 / 100, "Expected user1WantBalanceFinal > wantStartingAmount * 99 / 100");
+    }
+
+    function test_correctOwnerAndKeeper() external {
+        assertEq(vault.owner(), addressBookBeefy.vaultOwner, "Wrong vault owner.");
+        assertEq(strategy.owner(), addressBookBeefy.strategyOwner, "Wrong strategy owner.");
+        assertEq(strategy.keeper(), addressBookBeefy.keeper, "Wrong keeper.");
+    }
+
+    function test_harvestOnDeposit() external {
+        bool harvestOnDeposit = strategy.harvestOnDeposit();
+        if (harvestOnDeposit) {
+            console.log("Vault is harvestOnDeposit.");
+            assertEq(strategy.withdrawalFee(), 0, "Vault is harvestOnDeposit but has withdrawal fee.");
+        } else {
+            console.log("Vault is NOT harvestOnDeposit.");
+            assertEq(strategy.keeper(), addressBookBeefy.keeper, "Vault is not harvestOnDeposit but doesn't have withdrawal fee.");
```

Comment:
this does not check withdrawal fees 😄 but i figured it is copied from existing ProdTest